db/migrate/20170728140134_remove_email_from_enterprises.rb

```diff
@@ -0,0 +1,44 @@
+class RemoveEmailFromEnterprises < ActiveRecord::Migration
+  class Enterprise < ActiveRecord::Base; end
+  class Spree::User < ActiveRecord::Base; end
+
+  class EnterpriseRole < ActiveRecord::Base
+    belongs_to :user, class_name: 'Spree::User'
+    belongs_to :enterprise, class_name: 'Enterprise'
+  end
+
+  def up
+    Enterprise.reset_column_information
+    Spree::User.reset_column_information
+    EnterpriseRole.reset_column_information
+
+    Enterprise.select([:id, :email]).each do |enterprise|
+      contact_user = Spree::User.find_by_email enterprise.email
+      unless contact_user
+        password = Devise.friendly_token.first(8)
+        contact_user = Spree::User.create(email: enterprise.email, password: password, password_confirmation: password)
+        contact_user.send_reset_password_instructions
```

Comment:
This fails for invalid email addresses:
```
ArgumentError: job cannot be created for non-persisted record: #<Spree::User id: nil, encrypted_password: "f2cf19ad382abf84c97738ad4dd3331095b3d8ae8b124a02740...", password_salt: "CrPkqXhpNzGqJvGfAF91", email: "maya.bark-n-go and some spaces", remember_token: nil, persistence_token: nil, reset_password_token: nil, perishable_token: nil, sign_in_count: 0, failed_attempts: 0, last_request_at: nil, current_sign_in_at: nil, last_sign_in_at: nil, current_sign_in_ip: nil, last_sign_in_ip: nil, login: "maya.bark-n-go and some spaces", ship_address_id: nil, bill_address_id: nil, created_at: nil, updated_at: nil, authentication_token: nil, unlock_token: nil, locked_at: nil, remember_created_at: nil, spree_api_key: nil, reset_password_sent_at: nil, api_key: nil, enterprise_limit: 1, locale: nil, confirmation_token: nil, confirmed_at: nil, confirmation_sent_at: nil, unconfirmed_email: nil>
```

I got this running the migrations on our staging server. The `create` method saves the record only if validations pass. And if enterprise email is not valid, it can't create a delayed job to send the email.

Can we skip these enterprises? Something to discuss.

Reply:
I just had a chat with Rob and we think it would be best to set the current owner to receive notifications if we can't create a new user. That is a safe fallback.